File: core/play/src/main/scala/play/api/libs/typedmap/TypedEntry.scala
/*
 * Copyright (C) 2009-2019 Lightbend Inc. <https://www.lightbend.com>
 */

package play.api.libs.typedmap

/**
 * An entry that binds a typed key and a value. These entries can be
 * placed into a [[TypedMap]] or any other type of object with typed
 * values.
 *
 * @param key The key for this entry.
 * @param value The value for this entry.
 * @tparam A The type of the value.
 */
final case class TypedEntry[A](key: TypedKey[A], value: A) {
  /**
   * Convert the entry into a standard pair.
   */
  def toPair: (TypedKey[A], A) = (key, value)
}

File: core/play/src/main/java/play/libs/typedmap/TypedEntry.java
/*
 * Copyright (C) from 2022 The Play Framework Contributors <https://github.com/playframework>, 2011-2021 Lightbend Inc. <https://www.lightbend.com>
 */

package play.libs.typedmap;

/**
 * An entry that binds a typed key and a value. These entries can be placed into a {@link TypedMap}
 * or any other type of object with typed values.
 *
 * @param <A> The type of the key and value in this entry.
 */
public final class TypedEntry<A> {
  private final TypedKey<A> key;
  private final A value;

  public TypedEntry(TypedKey<A> key, A value) {
    this.key = key;
    this.value = value;
  }

  /**
   * @return the key part of this entry.
   */
  public TypedKey<A> key() {
    return key;
  }

  /**
   * @return the value part of this entry.
   */
  public A value() {
    return value;
  }

  /**
   * @return the Scala version for this entry.
   */
  public play.api.libs.typedmap.TypedEntry<A> asScala() {
    return new play.api.libs.typedmap.TypedEntry<>(this.key.asScala(), this.value);
  }
}
